sound/soc/sof/intel/hda.c

```diff
@@ -394,17 +423,33 @@ static int hda_init_caps(struct snd_sof_dev *sdev)
 			dev_info(bus->dev, "using HDA machine driver %s now\n",
 				 hda_mach->drv_name);
 
-			/* fixup topology file for HDMI only platforms */
-			if (codec_num == 1) {
-				/* use local variable for readability */
-				tplg_filename = pdata->tplg_filename;
-				tplg_filename = fixup_tplg_name(sdev, tplg_filename);
-				if (!tplg_filename) {
-					hda_codec_i915_exit(sdev);
-					return ret;
-				}
-				pdata->tplg_filename = tplg_filename;
+			if (codec_num == 1)
+				idisp_str = "-idisp";
+			else
+				idisp_str = "";
+
+			/* first check NHLT for DMICs */
+			dmic_num = check_nhlt_dmic(sdev);
+
+			/* allow for module parameter override */
+			if (hda_dmic_num)
+				dmic_num = hda_dmic_num;
+
+			if (dmic_num == 2)
+				dmic_str = "-2ch";
+			else if (dmic_num == 4)
+				dmic_str = "-4ch";
+			else
+				dmic_str = "";
```

Comment:
@plbossart actually an alternative would be, sincd dmic_num cant be anything other than 0, 2 or 4. 
if (dmic_num)
    dmic_str = kasprintf(GFP_KERNEL, "-%dch", dmic_num);
else
   dmic_str = "";

Reply:
hmm iddnt see the module param override, in which case if (dmic_num == 2 || dmic_num ==4)

Reply:
meh. I don't think it's required to allocate memory at all? Think of all the DAI names, we don't allocated anything and just point to constant strings.

Reply:
@plbossart hmm yes sorry I got confused there a bit. Agree no need to allocate memory.